src/placement/tests/benchmark_util.h

```diff
@@ -0,0 +1,216 @@
+/**
+ * (C) Copyright 2016-2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#ifndef __TIME_PROFILING_H__
+#define __TIME_PROFILING_H__
+
+#ifdef USE_TIME_PROFILING
+
+#include <stdio.h>
+#include <time.h>
+#include <float.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+
+#define COMPILER_BARRIER() asm volatile("" ::: "memory")
```

Comment:
(style)  Macros with complex values should be enclosed in parentheses

Reply:
In this case, wrapping (asm volatile... ) causes this code not to compile. It is not possible to comply with this style warning.

This instruction inserts a no-op assembly instruction that prevents the compiler from optimizing across this barrier (which would result in the benchmarking code not actually measuring the operation)